From f13fd6af4120b8b42d77377c9989457cda41899f Mon Sep 17 00:00:00 2001
From: Bruce Ashfield <bruce.ashfield@windriver.com>
Date: Sun, 15 Feb 2009 16:57:14 -0500
Subject: [PATCH 3/3] powerpc: add crtsavres.o to archprepare for kbuild

upstream commit da3de6df33f5f42ff9dc40093fbc884f524c9a49 adds
a dependency to all kernel modules to crtsavres. This
ensures that out-of-line register saves/restores work when
CONFIG_CC_OPTIMIZE_FOR_SIZE is enabled.

What it didn't do was ensure that it was built when
modules_prepare was run and hence out of tree powerpc kernel
modules can't properly link.

The fix is to add crtsavres.o to the archprepare rule in
the arch/powerpc/Makefile

Signed-off-by: Bruce Ashfield <bruce.ashfield@windriver.com>
---
 arch/powerpc/Makefile |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/arch/powerpc/Makefile b/arch/powerpc/Makefile
index c3811ab..ad16ff3 100644
--- a/arch/powerpc/Makefile
+++ b/arch/powerpc/Makefile
@@ -218,7 +218,7 @@ endif
 archclean:
 	$(Q)$(MAKE) $(clean)=$(boot)
 
-archprepare: checkbin
+archprepare: checkbin arch/powerpc/lib/crtsavres.o
 
 # Use the file '.tmp_gas_check' for binutils tests, as gas won't output
 # to stdout and these checks are run even on install targets.
-- 
1.6.5.2

